### ISSUE TITLE
[no-base-to-string] still many false positive with union values

### ISSUE THREAD

**Issue by HolgerJeromin** [typescript-eslint/typescript-eslint#1974]
Thanks for #1969  :-)
Found a real bug while testing no-base-to-string :-)

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-base-to-string": ["error"]
  }
}
```

```TS
((myvar1: string | number | boolean) => {
    myvar1.toString();
})('');
((myvar1: string | number | boolean | null) => {
    myvar1.toString();
})('');
((myvar1: number | boolean | null) => {
    myvar1.toString();
})(true);
```

**Expected Result**
should lint without problems
**Actual Result**
> warning @typescript-eslint/no-base-to-string : 'myvar1 may evaluate to '[object Object]' when stringified.

for all of them

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.31.0` |
| `@typescript-eslint/parser`        | `2.31.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
any union with `null` will fail, because `null` has no `toString` definition.
It was also specifically not handled on purpose, because `null` stringifies as `"null"`, which is not a good thing to leak into strings - you should be handling the null state.

If you are okay with `null` leaking into your strings, you can ignore it with the `ignoredTypeNames` option.

---

Unions are handled in this rule implicitly - the rule gets all definitions of the `toString` method for the type. This "just works" for union types, and returns one declaration for each type.

Unfortunately, the previously workaround that's been implemented for the boolean type doesn't implicitly work for unions...

**Comment by auhsoja** [comment]
Not sure if I handled this in the most technically-correct way. However, I've submitted a PR that should make the errors go away if merged.
